### ISSUE TITLE
Adding a querystring example for List<string>

### ISSUE THREAD

**Issue by sturlath** [domaindrivendev/Swashbuckle.AspNetCore#1676]
I want to be able to have example of  ```List<string> ``` for FromQuery string. 

I tried to use [Swashbuckle.AspNetCore.Filters](https://github.com/mattfrear/Swashbuckle.AspNetCore.Filters) and created an [issue there](https://github.com/mattfrear/Swashbuckle.AspNetCore.Filters/issues/152) 

@mattfrear has added the functionality [Add examples to XML params](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/pull/1629) that I would like to extend as soon as possible with a PR.

I was thinking something in the line of the following could work.

 ```Csharp
/// <summary>
/// GetByIds
/// </summary>
// <param name="Ids" example="[one,two,three]"></param>
public void GetByIds([FromQuery] List<string> ids)
{}
```

Would this work?

**Comment by mattfrear** [comment]
I like your idea to do a list [] rather than a delimiter `|`, I was thinking the same thing.
It would be worth looking through the XML spec to find out how to do multiple values for an XML attribute (if that is valid).

You will also need to get familiar with the OpenApi 3.0 spec and how multiple request querystring examples are specified (if that is possible). Try and get it working in https://editor.swagger.io/ before you start writing any code.

**Comment by sturlath** [comment]
Ok cool thanks. I got little lost in specs and documentation 🤦‍♂️

I'm not 100% I understand your question but in the [specification for adding examples](https://swagger.io/docs/specification/adding-examples/) they say 

> "Note that arrays and array items support single example but not multiple examples."

So if I got that right I think there is only room for one XML example node like I have made working here

![image](https://user-images.githubusercontent.com/2386572/82727361-89aeec00-9cd9-11ea-86ed-852ab983b99e.png)

Other interesting doc is [query paramters styles](https://swagger.io/docs/specification/serialization/)

Is this enough information to implement `<param name="Ids" example="[one,two,three]"></param>`  or is there something else I'm missing (I just use Swagger so what do I know 😆)?

**Comment by sturlath** [comment]
Hi @mattfrear I prepared some code and unit tests for it (just to be able to discuss further) and tried to push to a new branch but I get 403. Do I need permission or am I doing something wrong? 

![image](https://user-images.githubusercontent.com/2386572/82731787-ab6b9b80-9cf8-11ea-80d5-11424f94ba5f.png)

**Comment by mattfrear** [comment]
Looks like you're making pretty good progress.
I'd suggest that for 
> Array of string
`<param name="ByString" example="['one','two','three']"></param>`

should be OK. Unless you wanted to escape the double quotes with `&quot;` but that's not as readable.

It's up to @domaindrivendev to agree with what you're doing though, not me, as he is the owner of this repository.

**Comment by sturlath** [comment]
any feedback on this @domaindrivendev?

**Comment by DaleyKD** [comment]
Any update on this?  I'm very much looking forward to including this, but specifically for `long[]`.

Also, could/would this support `<returns example="['admin','user']">A string array of roles.</returns>` ?

**Comment by emilwall** [comment]
I'm also interested in using this, but for `List<Guid>`. Personally I don't have a strong preference between `'` and `&quot;`, I think the former looks better in the docstring whereas the latter looks better in the swagger GUI.

**Comment by domaindrivendev** [comment]
> any feedback on this @domaindrivendev?

If someone puts together a reasonable PR for this and it doesn't add too much complexity to the code base, I'd be happy to merge it in

**Comment by sturlath** [comment]
@domaindrivendev could you then comment on questions/design here above and in the [PR draft](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/pull/1677). I can finish this soon if I know that I'm headed in the right direction...

**Comment by domaindrivendev** [comment]
In latest master, this is now possible with the following syntax for top-level parameters:

```csharp
 /// <param name="tags" example='["foo", "bar"]'></param>
 [HttpGet]
 public IActionResult Get([FromQuery]List<string> tags)
```

Or the following for property-based parameters:

```csharp
public class QueryParams
{
    /// <example>["foo", "bar", "blah"]</example>
    public List<string> Tags { get; set; }
}
```

Will be officially available in the upcoming `6.0.0` release, but in the meantime you can try it out by pulling down the lastest preview build from [myget.org](https://myget.org/feed/domaindrivendev/package/nuget/Swashbuckle.AspNetCore).

**Comment by laventnc** [comment]
Is there something that I'm missing here... I have xml comments included (as described [here](https://github.com/domaindrivendev/Swashbuckle.AspNetCore#include-descriptions-from-xml-comments)).
However, with a setup similar to the following:
```
/// <param name="ids" example='[]'>blah</param>
[HttpPost]
public IActionResult Post(List<int> ids)
```

The resulting example & default value within the swagger page is:
```
[
  0
]
```

The generated xml looks normal:
```
<member name="M:HedgeFox.Api.Controllers.ParsersController.PostInvestorFunds(System.Collections.Generic.List{System.Int32})"> 
 <summary> ... </summary>
  <param name="ids" example="[]">blah.</param>
</member>
```
